mono/utils/mono-os-mutex.h

```diff
@@ -34,10 +34,7 @@
 
 G_BEGIN_DECLS
 
-#ifndef MONO_INFINITE_WAIT
```

Comment:
I'm confused about this change.  What if someone includes both `mono-os-mutex.h` and `mono-os-semaphore.h`? won't they get multiples `#define`s?

Reply:
As long as they are the same, it should be ok.

Reply:
The ifndef  pattern is not great because it allows varying defines.

Reply:
huh. Learned something about the C compiler/preprocessor today.

```
// clang -c -std=c99 -Wall -Wextra -Werror foo.c
#define X 1
#define X 1
int foo (void) { return X; }
```